lora/lorawan/region/au915.go

```diff
@@ -0,0 +1,22 @@
+//go:build lorawan_au915
```

Comment:
Build tags are probably not the best way to handle, since some radios do allow for setting the frequencies to support more than 1 region, obviously not at the same time.

Reply:
Radio modules can often handle multiple frequencies ranges, that's right, but onboard RF filters an antennas don't.

This is why you have two versions of nucleo WL55JC (stm32wl) that differs only by their radio filters :
![image](https://user-images.githubusercontent.com/561441/212486494-c7679704-27e9-4119-8b91-04ad85065804.png)

That said, some regions could run on the same radios:

![image](https://user-images.githubusercontent.com/561441/212486767-e94d443c-6961-48b4-be68-e9ce25632938.png)

Instead of tags, we could implement per-region struct that have a common interface:

The application would look something like this: 

```
lora.AddRegionSet(REGION_AS868, &RegionSetEU868{})
lora.AddRegionSet(REGION_AS868, &RegionSetAS923{})
lora.SetActiveRegionSet(REGION_AS868)
```

The developper can adds the needed regions, compatible with its hardware. 
And he can switch the region from its code. 

Olivier